### ISSUE TITLE
Log file location inconsistency on Desktop

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#7367]
### System details

    RStudio Edition : Desktop
    RStudio Version : started with 1.4.209 up through current build (1.4.603)
    OS Version      : macOS 10.15.5 (probably Linux desktop, too)
    R Version       : 3.6.3

### Steps to reproduce the problem
- Run desktop IDE
- Help / Diagnostics / Show Log Files
- Help / Diagnostics / Write Diagnostics Report

### Describe the problem in detail
- Show Log Files opens `~/.local/share/rstudio/log` folder
- The diagnostics report contains:

```
Log file: /Users/gary/.rstudio-desktop/log/rdesktop.log
--------------------------------------------------

(Not Found)

Log file: /Users/gary/.rstudio-desktop/log/rsession-gary.log
--------------------------------------------------

(Not Found)
```

### Describe the behavior you expected
Consistent location!

Looks like the diagnostics utility wasn't updated when we moved to standard folders. See https://github.com/rstudio/rstudio/blob/cb916717dfb9e628d3196da00b11da38204b90c8/src/cpp/diagnostics/DiagnosticsMain.cpp#L43

This discrepancy started with 1.4.209 which had commit e925e18c, 2020-04-16.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
Note, found three implementations in the desktop code:

SessionOptions.hpp:

```
core::FilePath userLogPath() const
{
   return userScratchPath().completeChildPath("log");
}
```

DiagnosticsMain.cpp

```
FilePath userLogPath()
{
   FilePath logPath = core::system::userSettingsPath(
      homePath(),
      "RStudio-Desktop"
   ).completeChildPath("log");
   return logPath;
}
```

DesktopUtils.cpp

```
FilePath userLogPath()
{
   FilePath userHomePath = core::system::userHomePath("R_USER|HOME");
   FilePath logPath = core::system::userSettingsPath(
      userHomePath,
      "RStudio-Desktop").completeChildPath("log");
   return logPath;
}
```

**Comment by kevinushey** [comment]
Is fixing this worthwhile for v1.3? (seems like it to me)

**Comment by kfeinauer** [comment]
Verified fixed RStudio 1.4.944
